src/viewer.js

```diff
@@ -267,25 +267,21 @@ $.Viewer = function( options ) {
     this.canvas.className = "openseadragon-canvas";
 
     // BEGIN: Visibility Redraw Observer
-    const self = this;
-
-    if (typeof IntersectionObserver !== 'undefined') {
+    if (typeof window.IntersectionObserver !== 'undefined') {
         const observer = new IntersectionObserver((entries) => {
             entries.forEach(entry => {
                 if (entry.isIntersecting) {
-                    const width = self.container.clientWidth;
-                    const height = self.container.clientHeight;
+                    const width = _this.container.clientWidth;
+                    const height = _this.container.clientHeight;
 
                     if (width > 0 && height > 0) {
-                        self.viewport.resize(width, height, true);
-                        self.forceRedraw();
-                        console.log("[OpenSeadragon] Auto-resized on visibility.");
+                        _this.forceResize();
                     }
                 }
             });
         }, { threshold: 0.1 });
 
-        observer.observe(this.container);
+        observer.observe(_this.container);
```

Comment:
This doesn't need to be a `_this` since it's not inside the closure of the `forEach` function.